packages/cli/src/commands/app/serve.ts

```diff
@@ -15,12 +15,40 @@
  */
 
 import fs from 'fs-extra';
+import chalk from 'chalk';
 import { Command } from 'commander';
 import { serveBundle } from '../../lib/bundler';
 import { loadCliConfig } from '../../lib/config';
 import { paths } from '../../lib/paths';
+import { Lockfile } from '../../lib/versioning';
+import { includedFilter } from '../versions/lint';
 
 export default async (cmd: Command) => {
+  const lockfile = await Lockfile.load(paths.resolveTargetRoot('yarn.lock'));
+  const result = lockfile.analyze({
+    filter: includedFilter,
+  });
+  const problemPackages = result.newVersions.map(({ name }) => name);
```

Comment:
It's best to warn about the `newRanges` result as well, as they will also cause breakages, although they can't be fixed without updating `package.json`s. That can be done with `yarn backstage-cli versions:bump` now though, but it can lead to breaking changes being pulled in that need fixing.

Reply:
Got it!

It looks like running `yarn backstage-cli versions:check` will check `newVersions` and `newRanges` and recommend either updating with `--fix` or updating manually in `package.json`. Would it be safe in this check to simply add the packages in `newRanges` to my existing problem packages array and display their names as well?

Reply:
Yep, has long as we keep the wording in sync with that, so perhaps "some of..."

Also realized that the only cause of trouble will be duplicate dependencies

Reply:
Added package `newRanges` to the check.